build/cmake/config.cmake

```diff
@@ -40,16 +40,26 @@ macro(wx_get_dependencies var lib)
                     get_target_property(dep_name ${dep} OUTPUT_NAME)
                 endif()
             else()
-                get_filename_component(dep_name ${dep} NAME)
+                # For the vaule like $<$<CONFIG:DEBUG>:LIB_PATH>
+                # Or $<$<NOT:$<CONFIG:DEBUG>>:LIB_PATH>
+                string(REGEX REPLACE "^.+>:(.+)>$" "\\1" dep_name ${dep})
+                if (NOT dep_name)
+                    set(dep_name ${dep})
+                endif()
```

Comment:
Why did you remove `get_filename_component(dep_name ${dep} NAME)`? We need this to get the library name.
With this change it will be `/Applications/Xcode.app/Contents/..../MacOSX11.3.sdk/usr/lib/libz.tbd` instead of  `-lz`, or `/usr/lib/x86_64-linux-gnu/libX11.so` instead of `-lX11`.

Reply:
For some port use `*Config.cmake` or `*-config.cmake`, the `INTERFACE_LINK_LIBRARIES` may contains cmake generator expression like `$<$<NOT:$<CONFIG:DEBUG>>:LIB_PATH>`.
If we use `get_filename_component(dep_name ${dep} NAME)`, the value of `dep_name` is `NAME>`.

Reply:
But can you use the regex to extract the library from the generator expression, and then use `get_filename_component` on this?

Reply:
I don't know why you use the file name directly here.
If the dependency library is not in system like _/usr/lib_, the compiler will fail to search this library.

Reply:
I try to generate the same output as `configure` does when generating `wx-config`. and it only has the library names.
If it is in a different path, maybe we have to specify `-Lpath/to/lib` before the library. Or require the user to have it in `PATH`.

Reply:
Yes, we must choose one way to solve this issue.